crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -629,6 +629,132 @@ impl Torus {
+        Self {
+            vertices: [a, b, c],
+        }
+    }
+
+    /// Get the area of the triangle
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+
+        let s = (ab + bc + ca) / 2.0;
+        (s * (s - ab) * (s - bc) * (s - ca)).sqrt()
+    }
+
+    /// Get the perimeter of the triangle
+    #[inline(always)]
+    pub fn perimeter(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+        ab + bc + ca
+    }
+
+    /// Get the normal of the triangle in the direction of the right-hand rule, assuming
+    /// the vertices are ordered in a counter-clockwise direction.
+    ///
+    /// The normal is computed as the cross product of the vectors `ab` and `ac`.
+    #[inline(always)]
+    pub fn normal(&self) -> Dir3 {
+        let [a, b, c] = self.vertices;
+        let ab = b - a;
+        let ac = c - a;
+        Dir3::new_unchecked(ab.cross(ac).normalize())
+    }
+
+    /// Reverse the order of the vertices in the triangle
+    #[inline(always)]
+    pub fn reverse(&mut self) {
+        self.vertices.swap(1, 2);
+    }
+
+    /// Get the centroid of the triangle
+    #[inline(always)]
+    pub fn centroid(&self) -> Vec3 {
+        (self.vertices[0] + self.vertices[1] + self.vertices[2]) / 3.0
+    }
+}
+
+impl Bounded3d for Triangle3d {
+    /// Get the bounding box of the triangle
+    fn aabb_3d(&self, translation: Vec3, rotation: glam::Quat) -> crate::bounding::Aabb3d {
+        let [a, b, c] = self.vertices;
+        let a = rotation * a + translation;
+        let b = rotation * b + translation;
+        let c = rotation * c + translation;
+
+        let min_x = a.x.min(b.x).min(c.x);
+        let min_y = a.y.min(b.y).min(c.y);
+        let min_z = a.z.min(b.z).min(c.z);
+        let max_x = a.x.max(b.x).max(c.x);
+        let max_y = a.y.max(b.y).max(c.y);
+        let max_z = a.z.max(b.z).max(c.z);
+
+        let bounding_center = Vec3::new(
+            (min_x + max_x) / 2.0,
+            (min_y + max_y) / 2.0,
+            (min_z + max_z) / 2.0,
+        );
+
+        let half_extents = Vec3::new(
+            (max_x - min_x) / 2.0,
+            (max_y - min_y) / 2.0,
+            (max_z - min_z) / 2.0,
+        );
+
+        crate::bounding::Aabb3d::new(bounding_center, half_extents)
+    }
+
+    /// Get the bounding sphere of the triangle
+    fn bounding_sphere(
+        &self,
+        translation: Vec3,
+        rotation: glam::Quat,
+    ) -> crate::bounding::BoundingSphere {
+        let [a, b, c] = self.vertices;
+        let a = rotation * a + translation;
+        let b = rotation * b + translation;
+        let c = rotation * c + translation;
+
+        let center = (a + b + c) / 3.0;
+        let radius = (a - center)
```

Comment:
Generally not the smallest bounding sphere, but it works as a bounding sphere, so that's good.

Reply:
If we added the `circumsphere` method, I think we could switch this to be implemented like `Triangle2d::bounding_circle`, which should give the minimum bounding sphere.

Reply:
Does a triangle's circumcircle always serve as its minimum bounding circle (MBC)? Given that the circumcircle's center is determined by the intersection of the triangle's side's perpendicular bisectors, the MBC coincides with the circumcircle only for triangles that are **not** obtuse.

<img width="634" alt="image" src="https://github.com/bevyengine/bevy/assets/19597793/733355f9-3a54-4ca3-ba0d-8d051aee50f8">

Reply:
For prior discussion, see https://github.com/bevyengine/bevy/pull/11336#discussion_r1451756007.

Reply:
For the `bounding_sphere` I will handle obtuse triangles separately so we get the **minimum** bounding sphere instead of just any bounding sphere.

I'm not aware of how other primitives are implementing it, but if we're always using the **minimum** bounding shape, then I suggest we change the function's name to `minimum_bounding_<shape>`. I like it because it's more noisy ([Stroustrup's Rule](https://www.thefeedbackloop.xyz/stroustrups-rule-and-layering-over-time/)).

Reply:
I don't think the API directly guarantees minimality (it sure doesn't produce minimal spheres for polygons IIRC).
But unless prohibitively expensive, we generally strive to have tight bounding boxes to make them maximally useful.